### ISSUE TITLE
undefined method `split' for nil:NilClass when tested with JRuby 9.0.5 and rails master branch

### ISSUE THREAD

**Issue by yahonda** [rsim/oracle-enhanced#1145]
#### Environment
- Oracle enhanced adapter master branch
- $ ruby -v
jruby 9.0.5.0 (2.2.3) 2016-01-26 7bee00d OpenJDK 64-Bit Server VM 25.111-b16 on 1.8.0_111-b16 +jit [linux-amd64]
- Rails master brach including https://github.com/rails/rails/commit/9360b6be63b7a452535699bcf6ae853df7f5eea7
- Oracle Database 11.2.0 Express Edition also Oracle Database 12.1.0.2 Enterprise Edition

It does NOT reproduce when any of these conditions are satisified:
- JRuby 9.1.6.0
- Use JRuby 9.0.4.0
- Revert https://github.com/rails/rails/commit/9360b6be63b7a452535699bcf6ae853df7f5eea7

```ruby
$ bundle exec rspec ./spec/active_record/connection_adapters/oracle_enhanced_schema_dump_spec.rb:180 

Coverage may be inaccurate; set the "--debug" command line option, or do JRUBY_OPTS="--debug" or set the "debug.fullTrace=true" option in your .jrubyrc
/home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/bundler/gems/simplecov-9912a0dfb479/lib/simplecov.rb:48: warning: tracing (e.g. set_trace_func) will not capture all events without --debug flag
==> Loading config from ENV or use default
==> Running specs with JRuby version 9.0.5.0
/home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/bundler/gems/ruby-plsql-439c68ae933c/lib/plsql/variable.rb:87: warning: shadowing outer local variable - column
==> Effective ActiveRecord version 5.1.0.alpha
Run options: include {:locations=>{"./spec/active_record/connection_adapters/oracle_enhanced_schema_dump_spec.rb"=>[180]}}
DEPRECATION WARNING: #table_exists? currently checks both tables and views. This behavior is deprecated and will be changed with Rails 5.1 to only check tables. Use #data_source_exists? instead. (called from schema_define at /home/yahonda/git/oracle-enhanced/spec/spec_helper.rb:112)
DEPRECATION WARNING: #table_exists? currently checks both tables and views. This behavior is deprecated and will be changed with Rails 5.1 to only check tables. Use #data_source_exists? instead. (called from schema_define at /home/yahonda/git/oracle-enhanced/spec/spec_helper.rb:112)
DEPRECATION WARNING: #table_exists? currently checks both tables and views. This behavior is deprecated and will be changed with Rails 5.1 to only check tables. Use #data_source_exists? instead. (called from schema_define at /home/yahonda/git/oracle-enhanced/spec/spec_helper.rb:112)
FDEPRECATION WARNING: #table_exists? currently checks both tables and views. This behavior is deprecated and will be changed with Rails 5.1 to only check tables. Use #data_source_exists? instead. (called from schema_define at /home/yahonda/git/oracle-enhanced/spec/spec_helper.rb:112)


Failures:

  1) OracleEnhancedAdapter schema dump foreign key constraints should include foreign key in schema dump
     Failure/Error: add_foreign_key :test_comments, :test_posts
     
     NoMethodError:
       undefined method `split' for nil:NilClass
     # /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb:13:in `accept'
     # /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb:78:in `visit_AddForeignKey'
     # /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb:27:in `block in visit_AlterTable'
     # org/jruby/RubyArray.java:2300:in `map'
     # /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb:27:in `visit_AlterTable'
     # /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb:14:in `accept'
     # /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb:922:in `add_foreign_key'
     # /home/yahonda/git/rails/activerecord/lib/active_record/migration.rb:851:in `block in method_missing'
     # /home/yahonda/git/rails/activerecord/lib/active_record/migration.rb:820:in `block in say_with_time'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/stdlib/benchmark.rb:288:in `measure'
     # /home/yahonda/git/rails/activerecord/lib/active_record/migration.rb:820:in `say_with_time'
     # /home/yahonda/git/rails/activerecord/lib/active_record/migration.rb:840:in `method_missing'
     # ./spec/active_record/connection_adapters/oracle_enhanced_schema_dump_spec.rb:182:in `block in (root)'
     # org/jruby/RubyBasicObject.java:1633:in `instance_eval'
     # ./spec/spec_helper.rb:114:in `block in schema_define'
     # /home/yahonda/git/rails/activerecord/lib/active_record/migration.rb:828:in `suppress_messages'
     # ./spec/spec_helper.rb:113:in `block in schema_define'
     # org/jruby/RubyBasicObject.java:1633:in `instance_eval'
     # /home/yahonda/git/rails/activerecord/lib/active_record/schema.rb:48:in `define'
     # /home/yahonda/git/rails/activerecord/lib/active_record/schema.rb:44:in `define'
     # ./spec/spec_helper.rb:112:in `schema_define'
     # ./spec/active_record/connection_adapters/oracle_enhanced_schema_dump_spec.rb:181:in `block in (root)'
     # org/jruby/RubyBasicObject.java:1670:in `instance_exec'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:254:in `block in run'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:496:in `block in with_around_and_singleton_context_hooks'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:453:in `block in with_around_example_hooks'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:464:in `block in run'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:602:in `run_around_example_hooks_for'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/hooks.rb:464:in `run'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:453:in `with_around_example_hooks'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:496:in `with_around_and_singleton_context_hooks'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example.rb:251:in `run'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example_group.rb:627:in `block in run_examples'
     # org/jruby/RubyArray.java:2300:in `map'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example_group.rb:623:in `run_examples'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example_group.rb:589:in `run'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example_group.rb:590:in `block in run'
     # org/jruby/RubyArray.java:2300:in `map'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/example_group.rb:590:in `run'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:113:in `block in run_specs'
     # org/jruby/RubyArray.java:2300:in `map'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:113:in `block in run_specs'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/configuration.rb:1835:in `with_suite_hooks'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:112:in `block in run_specs'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/reporter.rb:77:in `report'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:111:in `run_specs'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:87:in `run'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:71:in `run'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/lib/rspec/core/runner.rb:45:in `invoke'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/rspec-core-3.5.4/exe/rspec:4:in `<top>'
     # org/jruby/RubyKernel.java:955:in `load'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/bin/rspec:1:in `<top>'
     # org/jruby/RubyKernel.java:955:in `load'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/bin/rspec:23:in `(root)'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/lib/bundler/cli/exec.rb:1:in `(root)'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/lib/bundler/cli/exec.rb:74:in `kernel_load'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/lib/bundler/cli/exec.rb:27:in `run'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/lib/bundler/cli.rb:332:in `exec'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/lib/bundler/vendor/thor/lib/thor.rb:359:in `dispatch'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/lib/bundler/cli.rb:20:in `dispatch'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/lib/bundler/vendor/thor/lib/thor/base.rb:440:in `block in start'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/lib/bundler/cli.rb:11:in `start'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/lib/ruby/gems/shared/gems/bundler-1.13.7/exe/bundle:34:in `<top>'
     # org/jruby/RubyKernel.java:955:in `load'
     # /home/yahonda/.rbenv/versions/jruby-9.0.5.0/bin/bundle:23:in `<top>'

Finished in 2.76 seconds (files took 4.19 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/active_record/connection_adapters/oracle_enhanced_schema_dump_spec.rb:180 # OracleEnhancedAdapter schema dump foreign key constraints should include foreign key in schema dump

Coverage report generated for RSpec to /home/yahonda/git/oracle-enhanced/coverage. 0 / 1983 LOC (0.0%) covered.
$
```

**Comment by yahonda** [comment]
`ForeignKeyDefinition.new(name, to_table, options)` does not return struct name `ActiveRecord::ConnectionAdapters::ForeignKeyDefinition` with Ruby 9.0.5.0.  

* jruby 9.0.4.0 (2.2.2) 2015-11-12 b9fb7aa OpenJDK 64-Bit Server VM 25.111-b16 on 1.8.0_111-b16 +jit [linux-amd64]

```ruby
From: /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb @ line 406 ActiveRecord::ConnectionAdapters::AlterTable#add_foreign_key:

    404: def add_foreign_key(to_table, options)
    405:   require 'pry'
 => 406:   binding.pry
    407:   @foreign_key_adds << ForeignKeyDefinition.new(name, to_table, options)
    408: end

[1] pry(#<ActiveRecord::ConnectionAdapters::OracleEnhanced::AlterTable>)> @foreign_key_adds
=> []
[2] pry(#<ActiveRecord::ConnectionAdapters::OracleEnhanced::AlterTable>)> ForeignKeyDefinition.new(name, to_table, options)
=> #<struct ActiveRecord::ConnectionAdapters::ForeignKeyDefinition
 from_table="test_comments",
 to_table="test_posts",
 options={:column=>"test_post_id", :name=>"fk_rails_cc939fc91f"}>
[3] pry(#<ActiveRecord::ConnectionAdapters::OracleEnhanced::AlterTable>)> 
```

* jruby 9.0.5.0 (2.2.3) 2016-01-26 7bee00d OpenJDK 64-Bit Server VM 25.111-b16 on 1.8.0_111-b16 +jit [linux-amd64]

```ruby
From: /home/yahonda/git/rails/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb @ line 406 ActiveRecord::ConnectionAdapters::AlterTable#add_foreign_key:

    404: def add_foreign_key(to_table, options)
    405:   require 'pry'
 => 406:   binding.pry
    407:   @foreign_key_adds << ForeignKeyDefinition.new(name, to_table, options)
    408: end

[1] pry(#<ActiveRecord::ConnectionAdapters::OracleEnhanced::AlterTable>)> @foreign_key_adds
=> []
[2] pry(#<ActiveRecord::ConnectionAdapters::OracleEnhanced::AlterTable>)> ForeignKeyDefinition.new(name, to_table, options)
=> #<struct  from_table="test_comments", to_table="test_posts", options={:column=>"test_post_id", :name=>"fk_rails_cc939fc91f"}>
[3] pry(#<ActiveRecord::ConnectionAdapters::OracleEnhanced::AlterTable>)> 
```

* jruby 9.1.6.0 (2.3.1) 2016-11-09 0150a76 OpenJDK 64-Bit Server VM 25.111-b16 on 1.8.0_111-b16 +jit [linux-x86_64]

```ruby
    404: def add_foreign_key(to_table, options)
    405:   require 'pry'
 => 406:   binding.pry
    407:   @foreign_key_adds << ForeignKeyDefinition.new(name, to_table, options)
    408: end

[1] pry(#<ActiveRecord::ConnectionAdapters::OracleEnhanced::AlterTable>)> @foreign_key_adds
=> []
[2] pry(#<ActiveRecord::ConnectionAdapters::OracleEnhanced::AlterTable>)> ForeignKeyDefinition.new(name, to_table, options)
=> #<struct ActiveRecord::ConnectionAdapters::ForeignKeyDefinition
 from_table="test_comments",
 to_table="test_posts",
 options={:column=>"test_post_id", :name=>"fk_rails_cc939fc91f"}>
[3] pry(#<ActiveRecord::ConnectionAdapters::OracleEnhanced::AlterTable>)> 
```

**Comment by yahonda** [comment]
Closed by #1147.
